source/browseMode.py

```diff
@@ -1046,30 +1046,30 @@ def getControlFieldSpeech(self, attrs, ancestorAttrs, fieldType, formatConfig=No
 		textList = []
 		landmark = attrs.get("landmark")
 		if formatConfig["reportLandmarks"] and fieldType == "start_addedToControlFieldStack" and landmark:
-			try:
-				textList.append(attrs["name"])
-			except KeyError:
-				pass
-			if landmark == "region":
-				# The word landmark is superfluous for regions.
-				textList.append(aria.landmarkRoles[landmark])
-			else:
+			# Ensure that the name of the field gets presented even if normally it wouldn't. 
+			name=attrs.get('name')
+			if name and attrs.getPresentationCategory(ancestorAttrs,formatConfig,reason) is None:
+				textList.append(name)
+				if landmark == "region":
+					# The word landmark is superfluous for regions.
+					textList.append(aria.landmarkRoles[landmark])
+			if landmark != "region":
 				textList.append(_("%s landmark") % aria.landmarkRoles[landmark])
 		textList.append(super(BrowseModeDocumentTextInfo, self).getControlFieldSpeech(attrs, ancestorAttrs, fieldType, formatConfig, extraDetail, reason))
 		return " ".join(textList)
 
-	def getControlFieldBraille(self, field, ancestors, reportStart, formatConfig):
+	def getControlFieldBraille(self, field, ancestors, reportStart, formatConfig,reason=None):

```

Comment:
Is this extra reason kwarg ever used and different than None? I'd say you can also just pass reason=None to field.getPresentationCategory